### PR TITLE
Type removal for 2.0

### LINKED ISSUE #217: Breaking changes for 2.0
Get started with incorporating the breaking changes for 2.0 from OpenSearch 2.0.

To do so:
1. Branch off a 1.x from main if not already done so that development for 2.0 can start on main.
2. Incorporate breaking changes as mentioned in https://github.com/opensearch-project/opensearch-clients/issues/12.
3. Add integ tests support to run against unreleased OpenSearch.

Example PR from java-client: https://github.com/opensearch-project/opensearch-java/pull/132

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
This PR adds breaking changes from OpenSearch 2.0. The changes included are type removal and include-type-name parameter removal. 
 
### Issues Resolved
#217 
 
### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ananzh — CHANGES_REQUESTED** [review]
Hey @VachaShah , thanks a lot for contributing. Regarding this PR, I see our major change is to get ride of `_type`, `existsType/exists_type`, ``flush_synced/flushSynced` apis. I am completely fine with `_type` and `existsType/exists_type` api since OS has removed them. However, for `flush_synced/flushSynced` api, I see they still exist in OS 2.0. 

Meanwhile from what I test on these APIs:
```
'use strict';
var host = 'localhost';
var protocol = 'https';
var port = 9200;
var auth = 'admin:admin'; // For testing only. Don't store credentials in code.

var { Client } = require('@opensearch-project/opensearch');
var fs = require('fs');
var client = new Client({
    node: 'http://localhost:9200',
 }
})
async function search() {
    // Create an index with non-default settings.
    var index_name = 'books'
    var settings = {
        'settings': {
            'index': {
                'number_of_shards': 4,
                'number_of_replicas': 3
            }
        }
    }
    var response = await client.indices.create({
        index: index_name,
        body: settings
    })
    console.log('Creating index:')
    console.log(response.body)
    

    var response = await client.indices.exists_type({ index: "books", type: "book" });
    console.log('Exist type:')
    console.log(response.body)

    var response = await client.indices.flush({ index: "books"});
    console.log('Flush:')
    console.log(response.body)

    var response = await client.indices.flushSynced({ index: "books"});
    console.log('Flush synced:')
    console.log(response.body)
 }
search().catch(console.log)

```

`existsType/exists_type` will report an error as expected. 


However, `flush_synced/flushSynced` still has responses:
```
~/work/test$ node data.js
Creating index:
{ acknowledged: true, shards_acknowledged: true, index: 'books' }
Flush synced:
{ _shards: { total: 16, successful: 4, failed: 0 } }
```
To sum up, my question is 1) did we completely removed `flush_synced/flushSynced`? 2) if not, can we just mark them as deprecated but not remove?

**Comment by VachaShah** [comment]
> Hey @VachaShah , thanks a lot for contributing. Regarding this PR, I see our major change is to get ride of `_type`, `existsType/exists_type`, ``flush_synced/flushSynced`apis. I am completely fine with`_type` and `existsType/exists_type` api since OS has removed them. However, for `flush_synced/flushSynced` api, I see they still exist in OS 2.0.
> 
> Meanwhile from what I test on these APIs:
> 
> ```
> 'use strict';
> var host = 'localhost';
> var protocol = 'https';
> var port = 9200;
> var auth = 'admin:admin'; // For testing only. Don't store credentials in code.
> 
> var { Client } = require('@opensearch-project/opensearch');
> var fs = require('fs');
> var client = new Client({
>     node: 'http://localhost:9200',
>  }
> })
> async function search() {
>     // Create an index with non-default settings.
>     var index_name = 'books'
>     var settings = {
>         'settings': {
>             'index': {
>                 'number_of_shards': 4,
>                 'number_of_replicas': 3
>             }
>         }
>     }
>     var response = await client.indices.create({
>         index: index_name,
>         body: settings
>     })
>     console.log('Creating index:')
>     console.log(response.body)
>     
> 
>     var response = await client.indices.exists_type({ index: "books", type: "book" });
>     console.log('Exist type:')
>     console.log(response.body)
> 
>     var response = await client.indices.flush({ index: "books"});
>     console.log('Flush:')
>     console.log(response.body)
> 
>     var response = await client.indices.flushSynced({ index: "books"});
>     console.log('Flush synced:')
>     console.log(response.body)
>  }
> search().catch(console.log)
> ```
> 
> `existsType/exists_type` will report an error as expected.
> 
> However, `flush_synced/flushSynced` still has responses:
> 
> ```
> ~/work/test$ node data.js
> Creating index:
> { acknowledged: true, shards_acknowledged: true, index: 'books' }
> Flush synced:
> { _shards: { total: 16, successful: 4, failed: 0 } }
> ```
> 
> To sum up, my question is 1) did we completely removed `flush_synced/flushSynced`? 2) if not, can we just mark them as deprecated but not remove?

@ananzh Thanks for the review! Yes the deprecated `indices.flush_synced` api was completely removed from OpenSearch in commit https://github.com/opensearch-project/OpenSearch/commit/0791c88af1bee78dee60902c1d5f1346d5d64517

**Comment by ananzh** [comment]
@VachaShah Thanks for the reply. Two more questions:

1)When I use OpenSearch 2.0 to test, I can see the responses from flush and flushSync are the same. 

```
Creating index:
{ acknowledged: true, shards_acknowledged: true, index: 'books' }
Flush:
{ _shards: { total: 16, successful: 4, failed: 0 } }
Flush synced:
{ _shards: { total: 16, successful: 4, failed: 0 } }
```


This is different from the result of OpenSearch 1.3:
```
Flush synced:
{ _shards: { total: 16, successful: 4, failed: 0 },
  books: { total: 16, successful: 4, failed: 0 } }
```
It seems to me that OpenSearch transfers the flushSync request to regular flush. However, I could not locate from OpenSearch repo where the transfer is made. Could you point it to me?

2) OpenSearch 2.0 removes both `indices.exists_type` and `indices.flushSynced`. However, `indices.exists_type` fails directly (see below test result) while `indices.flushSynced` is still working. Do you know why we treat two removed APIs differently?

```
Creating index:
{ acknowledged: true, shards_acknowledged: true, index: 'books' }
{ ResponseError: Response Error
    at onBody (/home/anan/work/opensearch-js/lib/Transport.js:374:23)
    at IncomingMessage.onEnd (/home/anan/work/opensearch-js/lib/Transport.js:293:11)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  name: 'ResponseError',
  meta:
   { body: '',
     statusCode: 400,
     headers:
      { 'content-type': 'application/json; charset=UTF-8',
        'content-length': '77' },
     meta:
      { context: null,
        request: [Object],
        name: 'opensearch-js',
        connection: [Object],
        attempts: 0,
        aborted: false } } }
```

**Comment by VachaShah** [comment]
> @VachaShah Thanks for the reply. Two more questions:
> 
> 1. When I use OpenSearch 2.0 to test, I can see the responses from flush and flushSync are the same.
> 
> ```
> Creating index:
> { acknowledged: true, shards_acknowledged: true, index: 'books' }
> Flush:
> { _shards: { total: 16, successful: 4, failed: 0 } }
> Flush synced:
> { _shards: { total: 16, successful: 4, failed: 0 } }
> ```
> 
> This is different from the result of OpenSearch 1.3:
> 
> ```
> Flush synced:
> { _shards: { total: 16, successful: 4, failed: 0 },
>   books: { total: 16, successful: 4, failed: 0 } }
> ```
> 
> It seems to me that OpenSearch transfers the flushSync request to regular flush. However, I could not locate from OpenSearch repo where the transfer is made. Could you point it to me? or if my understanding is not correct, then why flushSync is still working using OpenSearch 2.0?
> 
> 2. OpenSearch 2.0 removes both `indices.exists_type` and `indices.flushSynced`. However, `indices.exists_type` fails directly (see below test result) while `indices.flushSynced` is still working. Do you know why we treat two removed APIs differently?
> 
> ```
> Creating index:
> { acknowledged: true, shards_acknowledged: true, index: 'books' }
> { ResponseError: Response Error
>     at onBody (/home/anan/work/opensearch-js/lib/Transport.js:374:23)
>     at IncomingMessage.onEnd (/home/anan/work/opensearch-js/lib/Transport.js:293:11)
>     at IncomingMessage.emit (events.js:203:15)
>     at endReadableNT (_stream_readable.js:1145:12)
>     at process._tickCallback (internal/process/next_tick.js:63:19)
>   name: 'ResponseError',
>   meta:
>    { body: '',
>      statusCode: 400,
>      headers:
>       { 'content-type': 'application/json; charset=UTF-8',
>         'content-length': '77' },
>      meta:
>       { context: null,
>         request: [Object],
>         name: 'opensearch-js',
>         connection: [Object],
>         attempts: 0,
>         aborted: false } } }
> ```

Not sure why the response between the removed apis is different, but since flush synced is completely removed from OpenSearch, we are removing support for that api.

**Review by ananzh — APPROVED** [review]
LGTM

**Review by CEHENKLE — APPROVED** [review]
